[WIP] Terraform Docs for TF policy (#2690)
hashicorp/web-unified-docs: content/terraform-docs-common/docs/cloud-docs/workspaces/policy-enforcement/manage-policy-sets/configure.mdx

When your workspace exeuction mode is also set to **Agent**, HCP Terraform and Terraform Enterprise use the same agent instance to evaluate policies and run Terraform operations. When the workspace execution mode is set to **Local** or **Remote**, HCP Terraform or Terraform Enterprise starts a new agent to evaluate the policy.

You can explicitly set your workspace execution mode to **Agent** or configure your workspace to inherit an execution mode from [project settings](/terraform/cloud-docs/projects#execution-mode) or [organization settings](/terraform/cloud-docs/users-teams-organizations/organizations/settings).

Use HCP Terraform Agent mode to execute your policy set using a specific version of Sentinel. Because your policies run in an isolated environment, you can also use this mode to offload resource consumption from HCP Terraform or Terraform Enterprise. 

HCP Terraform Agent mode can result in slower startup times because of the time necessary to dispatch an agent or start the control plane. This mode cannot access cost estimation data.

> To evaluate Terraform policy policies, workspaces running in "Agent" execution mode must be of a minimum version 1.28.10.

Reviewer: ```suggestion
When the HCP Terraform execution mode is set to **Agent**, you must connect to HCP Terraform agents v1.28.10 or newer to evaluate policies written for Terraform policy.
```
We should move this to the Requirements section on this page.

HCP Terraform Agent mode can result in slower startup times because of the time necessary to dispatch an agent or start the control plane. This mode cannot access cost estimation data.

> To evaluate Terraform policy policies, workspaces running in "Agent" execution mode must be of a minimum version 1.28.10.

Reviewer 2: Workspaces running in 'local agent'/'private agent pools' have this version constraint. If the workspace is using the common HCP TF agent pool I think they are automatically on latest.

[Sentinel parameters](/sentinel/docs/language/parameters) are a list of key-value pairs that HCP Terraform and Terraform Enterprise send to the Sentinel runtime when performing policy checks on workspaces. If the value parses as JSON, HCP Terraform and Terraform Enterprise send it to Sentinel as the corresponding type, such as string, Boolean, integer, map, or list. If the value fails JSON validation, the value is sent as a string.

You can set Sentinel parameters when you [edit a policy set](#edit-policy-sets).

### Terraform policy inputs

[Terraform policy inputs](/terraform/policy/write/inputs-and-locals) are a list of key-value pairs that HCP Terraform and Terraform Enterprise send to the Terraform policy runtime as parameters when performing policy checks on workspaces. If the value parses as JSON, HCP Terraform and Terraform Enterprise send it to Terraform policy as the corresponding type, such as string, Boolean, integer, map, or list. If the value fails JSON validation, the value is sent as a string.

> You can set Terraform policy parameters when you [edit a policy set](#edit-policy-sets). Terraform policy will also load values for inputs from workspace environment variables with a matching name. For example, a workspace with an environment variable of `instance_type` would set the value of the `instance_type` input.

Reviewer: ```suggestion
You can set Terraform policy parameters when you [edit a policy set](#edit-policy-sets). Terraform policy also loads values for inputs from workspace environment variables with a matching name. For example, a workspace with an environment variable of `instance_type` sets the value of the `instance_type` input.
```

> When Terraform policy evaluates your Terraform runs, it will set values for inputs in your policies through the following methods, in order of precendence:

Reviewer: ```suggestion
When Terraform policy evaluates your Terraform runs, it sets values for inputs in your policies through the following methods in order of precedence:
```

[Sentinel parameters](/sentinel/docs/language/parameters) are a list of key-value pairs that HCP Terraform and Terraform Enterprise send to the Sentinel runtime when performing policy checks on workspaces. If the value parses as JSON, HCP Terraform and Terraform Enterprise send it to Sentinel as the corresponding type, such as string, Boolean, integer, map, or list. If the value fails JSON validation, the value is sent as a string.

You can set Sentinel parameters when you [edit a policy set](#edit-policy-sets).

### Terraform policy inputs

[Terraform policy inputs](/terraform/policy/write/inputs-and-locals) are a list of key-value pairs that HCP Terraform and Terraform Enterprise send to the Terraform policy runtime as parameters when performing policy checks on workspaces. If the value parses as JSON, HCP Terraform and Terraform Enterprise send it to Terraform policy as the corresponding type, such as string, Boolean, integer, map, or list. If the value fails JSON validation, the value is sent as a string.

> You can set Terraform policy parameters when you [edit a policy set](#edit-policy-sets). Terraform policy will also load values for inputs from workspace environment variables prefixed with `TF_VAR_`. For example, a workspace with an environment variable of `TF_VAR_instance_type` would set the value of the `instance_type` input.

Reviewer 3: No need to have TF_VAR_ prefix for workspace env variables for policy input. Instead refer the key `instance_type` directly in workspace environment variable.
cc: @vpaul97 @viveksinghgit95
